src/chemistry_functionality.jl

```diff
@@ -6,93 +7,195 @@ Symbolics.option_to_metadata_type(::Val{:iscompound}) = CompoundSpecies
 Symbolics.option_to_metadata_type(::Val{:components}) = CompoundComponents
 Symbolics.option_to_metadata_type(::Val{:coefficients}) = CompoundCoefficients
 
-macro compound(species_expr, arr_expr...)
-    # Ensure the species name is a valid expression
-    if !(species_expr isa Expr && species_expr.head == :call)
-        error("Invalid species name in @compound macro")
+### Create @compound Macro(s) ###
+
+"""
+    @compound
+
+Macro that creates a compound species, which is composed of smaller component species.
+
+Example:
+```julia
+@variables t
+@species C(t) O(t)
+@compound CO2(t) = C + 2O
+```
+
+Notes: 
+- The component species must be defined before using the `@compound` macro.
+"""
+macro compound(expr)
+    make_compound(MacroTools.striplines(expr))
+end
+
+# Declares compound error messages:
+const COMPOUND_CREATION_ERROR_BASE = "Malformed input to @compound. Should use form like e.g. \"@compound CO2 ~ C + 2O\"."
+const COMPOUND_CREATION_ERROR_BAD_SEPARATOR = "Malformed input to @compound. Left-hand side (the compound) and the right-hand side (the components) should be separated by a \"~\" (e.g. \"@compound CO2 ~ C + 2O\"). If the left hand side contains metadata and/or default values, this should be enclosed by \"()\" (e.g. \"@compound (CO2 = 1.0, [output=true]) ~ C + 2O\")."
+const COMPOUND_CREATION_ERROR_DEPENDENT_VAR_REQUIRED = "When the components (collectively) have more than 1 independent variable, independent variables have to be specified for the compound (e.g. `@compound CO2(t,s) ~ C + 2O`). This is required since the @compound macro cannot infer the correct order of the independent variables."
```

Comment:
Is this still true or do you infer it as the union? In the tutorial you say you shouldn't give arguments to the compound I think.

Reply:
Arguments are required, we couldn't do the union because we wouldn't be able to infer the correct order of the ivs. Will double read through the tutorial to ensure this is written there.

Reply:
added an extra bit about this.